omod/src/main/java/org/openmrs/module/coreapps/page/controller/systemadministration/SystemAdministrationPageController.java

```diff
@@ -17,6 +18,11 @@ public void controller(PageModel model, UiSessionContext emrContext,
                            @SpringBean("appFrameworkService") AppFrameworkService appFrameworkService) {
 
         emrContext.requireAuthentication();
+		
+		if (!emrContext.getCurrentUser().getPrivileges().toString().contains("coreapps.systemAdministration")
+		        && (!emrContext.getCurrentUser().isSuperUser())) {
```

Comment:
Thanks @CrooksDC ,looks like using && operator locks out the other scenarios we want for this,ie we need to make sure that if one doesn't have the system administration privilege's we throw an exception or  incase one is not a super user we throw an exception too, why not use the ||  operator?

Reply:
I think it's supposed to be && because you want both of those to be true to throw an exception. If only one is true and you are using the || it will go and throw the expectation even if one is false. Like if it does not have the privilege but it is a superuser and I was using the || it would go into the if statement and throw the exception even though I was a superuser.
@HerbertYiga